tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/Reordering.scala

```diff
@@ -0,0 +1,50 @@
+package at.forsyte.apalache.tla.assignments
+
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.lir.{LetInEx, OperEx, TlaEx, TlaOperDecl}
+import at.forsyte.apalache.tla.lir.transformations.{TlaExTransformation, TransformationTracker}
+
+class Reordering[T]( ord : Ordering[T], rankingFn : TlaEx => T, tracker : TransformationTracker ) {
+
+  def reorder : TlaExTransformation = tracker.track {
```

Comment:
Seems like we might benefit from a unit test or two for this method. Does that seem reasonable?

Reply:
I can add some, but this entire class is basically just boilerplate around line 12.

Reply:
You're definitely a better judge than I am of where unit tests would be applicable, but it does seem to me intuitively that we should have *some* kind of unit tests to ensure the reordering is happening as we expect. Does that seem sensible?